Cubical/M-types/M-type/Base.agda

```diff
@@ -0,0 +1,229 @@
+    -> let (A , B) = S in
+    Iso
+      (Σ A (λ a → B a → M-type S))
+      (Σ A (λ a → Σ ((n : ℕ) → B a → X (sequence S) n) λ u → (n : ℕ) → π (sequence S) ∘ (u (suc n)) ≡ u n))
+fst (fun (sym-α-iso-step-6 {S = S@(A , B)}) (x , y')) = x
+snd (fun (sym-α-iso-step-6 {S = S@(A , B)}) (x , y')) = transport (sym ((λ (a : A) → sym (lemma10 (B a , (λ x → a , (λ x₁ → x₁))))) x)) y'
+
+fst (inv (sym-α-iso-step-6 {S = S@(A , B)}) (x , y)) = x
+snd (inv (sym-α-iso-step-6 {S = S@(A , B)}) (x , y)) = transport ((λ (a : A) → sym (lemma10 (B a , (λ x → a , (λ x₁ → x₁))))) x) y
+
+rightInv (sym-α-iso-step-6 {S = S@(A , B)}) (x , y') = ΣPathP (refl {x = x} , transport⁻Transport ((λ (a : A) → sym (lemma10 (B a , (λ x → a , (λ x₁ → x₁))))) x) y')
+
+leftInv (sym-α-iso-step-6 {S = S@(A , B)}) (x , y) = ΣPathP (refl {x = x} , transportTransport⁻ ((λ (a : A) → sym (lemma10 (B a , (λ x → a , (λ x₁ → x₁))))) x) y)
+
+α-iso-step-6 : ∀ {ℓ} {S : Container {ℓ}}
+    -> let (A , B) = S in
+    Σ A (λ a → Σ ((n : ℕ) → B a → X (sequence S) n) λ u → (n : ℕ) → π (sequence S) ∘ (u (suc n)) ≡ u n)
+    ≡ Σ A (λ a → B a → M-type S)
+α-iso-step-6 {S = S@(A , B)} = Σ-ap-iso₂ (λ a i → lemma10 (B a , (λ x → a , (λ x₁ → x₁))) (~ i))
+
+-- Lemma 13
+α-iso : ∀ {ℓ} {S : Container {ℓ}} -> L (PX,Pπ S) ≡ P₀ {S = S} (M-type S) -- L^P ≡ PL
+α-iso {S = S@(A , B)} =
+  α-iso-step-1-4 □ (α-iso-step-5 □ α-iso-step-6)
+
+-----------------------------------------------------
+-- Shifting the limit of a chain is an equivalence --
+-----------------------------------------------------
+
+comp-α-iso-step-1-4-Iso-Sym-L-unique-iso : ∀ {ℓ} {S : Container {ℓ}} -> let (A , B) = S in Iso (Σ (Σ ((n : ℕ) → A) (λ a → (n : ℕ) → a (suc n) ≡ a n)) λ a → Σ ((n : ℕ) → B (a .fst n) → X (sequence S) n) λ u → (n : ℕ) → PathP (λ x → B (a .snd n x) → X (sequence S) n) (π (sequence S) ∘ u (suc n)) (u n))  (L (sequence S))
+fst (fun comp-α-iso-step-1-4-Iso-Sym-L-unique-iso (a , b)) 0 = lift tt
+fst (fun comp-α-iso-step-1-4-Iso-Sym-L-unique-iso (a , b)) (suc n) = (a .fst n) , (b .fst n)
+snd (fun comp-α-iso-step-1-4-Iso-Sym-L-unique-iso (a , b)) 0 = refl {x = lift tt}
+snd (fun comp-α-iso-step-1-4-Iso-Sym-L-unique-iso (a , b)) (suc n) i = a .snd n i , b .snd n i
+fst (fst (inv comp-α-iso-step-1-4-Iso-Sym-L-unique-iso x)) n = (x .fst) (suc n) .fst
+snd (fst (inv comp-α-iso-step-1-4-Iso-Sym-L-unique-iso x)) n i = (x .snd) (suc n) i .fst
+fst (snd (inv comp-α-iso-step-1-4-Iso-Sym-L-unique-iso x)) n = (x .fst) (suc n) .snd
+snd (snd (inv comp-α-iso-step-1-4-Iso-Sym-L-unique-iso x)) n i = (x .snd) (suc n) i .snd
+fst (rightInv comp-α-iso-step-1-4-Iso-Sym-L-unique-iso (b , c) i) 0 = lift tt
+fst (rightInv comp-α-iso-step-1-4-Iso-Sym-L-unique-iso (b , c) i) (suc n) = refl i
+snd (rightInv comp-α-iso-step-1-4-Iso-Sym-L-unique-iso (b , c) i) 0 = refl
+snd (rightInv comp-α-iso-step-1-4-Iso-Sym-L-unique-iso (b , c) i) (suc n) = c (suc n)
+leftInv comp-α-iso-step-1-4-Iso-Sym-L-unique-iso a = ΣPathP (refl , refl)
```

Comment:
I guess this is what you mean by unfolded "a bit much"?

Did you try with just `fun`, `inv`, `leftInv` and `rightInv` as copatterns and the rest as lemmas? Was that still too slow?

Reply:
In most cases it was enough, but I had to expand in one or two cases, so I just did it to be sure it would work. I will return it to a more readable form.

Reply:
It now looks like this:

```
comp-α-iso-step-1-4-Iso-Sym-L-unique-iso : ∀ {ℓ} {S : Container {ℓ}} -> let (A , B) = S in Iso (Σ (Σ ((n : ℕ) → A) (λ a → (n : ℕ) → a (suc n) ≡ a n)) λ a → Σ ((n : ℕ) → B (a .fst n) → X (sequence S) n) λ u → (n : ℕ) → PathP (λ x → B (a .snd n x) → X (sequence S) n) (π (sequence S) ∘ u (suc n)) (u n))  (L (sequence S))
fun comp-α-iso-step-1-4-Iso-Sym-L-unique-iso (a , b) = (λ { 0 → lift tt ; (suc n) → (a .fst n) , (b .fst n)}) , λ { 0 → refl {x = lift tt} ; (suc m) i → a .snd m i , b .snd m i }
inv comp-α-iso-step-1-4-Iso-Sym-L-unique-iso x = ((λ n → (x .fst) (suc n) .fst) , λ n i → (x .snd) (suc n) i .fst) , (λ n → (x .fst) (suc n) .snd) , λ n i → (x .snd) (suc n) i .snd
fst (rightInv comp-α-iso-step-1-4-Iso-Sym-L-unique-iso (b , c) i) 0 = lift tt
fst (rightInv comp-α-iso-step-1-4-Iso-Sym-L-unique-iso (b , c) i) (suc n) = refl i
snd (rightInv comp-α-iso-step-1-4-Iso-Sym-L-unique-iso (b , c) i) 0 = refl
snd (rightInv comp-α-iso-step-1-4-Iso-Sym-L-unique-iso (b , c) i) (suc n) = c (suc n)
leftInv comp-α-iso-step-1-4-Iso-Sym-L-unique-iso a = ΣPathP (refl , refl) 
```

I needed to match on `0` and `(suc n)` as pattern/copattern to make `rightInv` type check.